drivers/storage/portworx/component/telemetry.go

```diff
@@ -813,39 +814,26 @@ func (t *telemetry) createDeploymentPhonehomeCluster(
 		},
 	}}
 
-	// Count number of replicas using number of storage node on this cluster
-	t.sdkConn, err = pxutil.GetPortworxConn(t.sdkConn, t.k8sClient, cluster.Namespace)
-	if err != nil {
-		return err
-	}
-
-	replicasCount, err := pxutil.CountStorageNodes(cluster, t.sdkConn, t.k8sClient)
-	if err != nil {
-		t.closeSdkConn()
-		return err
-	}
-	num := int32(replicasCount)
-	deployment.Spec.Replicas = &num
-	deployment.Spec.Template.Spec.ServiceAccountName = ServiceAccountNamePxTelemetry
-	pxutil.ApplyStorageClusterSettings(cluster, deployment)
+	daemonset.Spec.Template.Spec.ServiceAccountName = ServiceAccountNamePxTelemetry
+	pxutil.ApplyStorageClusterSettingsToPodSpec(cluster, &daemonset.Spec.Template.Spec)
 
-	existingDeployment, err := k8sutil.GetDeployment(t.k8sClient, deployment)
-	if err != nil {
+	existingDaemonSet := &appsv1.DaemonSet{}
+	if err := k8sutil.GetDaemonSet(t.k8sClient, daemonset.Name, daemonset.Namespace, existingDaemonSet); err != nil {
```

Comment:
if the daemonset does not exist, we should not fail, you can refer to telemetry-java.go

and when doing deepEqual we shall handle the nil daemonset when it does not exist. so I was thinking about still have DeepEqualDaemonset and DeepEqualDeployment that checks nil then call DeepEqualPodTemplate.

Reply:
yes, I've verified that. when it doesn't exist, the result won't be nil but empty object, so deep equal will return false.